### DISCUSSION TITLE
Prometheus remote_write wont work

### DISCUSSION THREAD

**Discussion by Akhofreeze** [grafana/mimir#5405] [Help and support]
Hi ! 
I have an issue when I try to use the "remote_write" on prometheus.

I got this error: 
`ts=2023-07-03T15:22:05.042Z caller=dedupe.go:112 component=remote level=warn remote_name=b46aea url=mimir.namespace.svc:9009 msg="Failed to send batch, retrying" err="Post \"mimir.namespace.svc:9009\": unsupported protocol scheme \"mimir.namespace.svc\""
`

No logs appear on mimir.
Here is my Prometheus config file:
```
apiVersion: v1
data:
  prometheus.yml: |-
    global:
      evaluation_interval: 1m
      scrape_interval: 1m
      scrape_timeout: 30s
    remote_write:
      - url: mimir.namespace.svc:9009
        tls_config:
          insecure_skip_verify: true
        headers:
          X-Scope-OrgID: dev
  
kind: ConfigMap
metadata:
  creationTimestamp: "2022-03-13T21:43:45Z"
  labels:
    name: prometheus-tec-conf
  name: prometheus-tec-conf
  namespace: namespaceName
  resourceVersion: "869440854"
  uid: 2ca75faf-5119-4472-8fe5-106ab2ed440e
```

**Comment by charleskorn** [comment]
Have you tried adding `http://` or `https://` to the start of the value for `url` (eg. `url: http://mimir.namespace.svc:9009`)? Looks like Prometheus is interpreting `mimir.namespace.svc` as the protocol.

**Reply by Akhofreeze** [reply]
Hello ! 

Yes I've tried both `http://` and `https://` and I tried also to use an `Ingress` and tried both protocols too.

**Reply by charleskorn** [reply]
Do you get the same error with `http` or `https`? Or something different?

**Reply by Akhofreeze** [reply]
Ok it's now working, the changes was : 
Use the cluster.local service url: 
`http://mimir.namespace.svc.cluster.local:9009/api/v1/push`
And force Mimir to listen on 9009 port with `http_listen_port: 9009`
